src/main/java/org/folio/circulation/services/ItemsInTransitReportService.java

```diff
@@ -79,9 +84,30 @@ public CompletableFuture<Result<JsonObject>> buildReport() {
       .thenCompose(r -> r.after(this::fetchUsers))
       .thenCompose(r -> r.after(this::fetchPatronGroups))
       .thenCompose(r -> r.after(this::fetchServicePoints))
-      .thenApply(this::mapToJsonObject);
+      .thenApply(this::mapToJsonObject)
+      .whenComplete(this::handleResult)
+      .exceptionally(this::handleException);
+  }
+
+  private void handleResult(Result<JsonObject> result, Throwable throwable) {
+    if (result.succeeded()) {
+      log.info("The report was built successfully");
+    } else if (throwable != null) {
+      log.error("Report failed {}, {}, {}, {}", result.cause(), throwable.getCause(),
+        throwable.getMessage(), throwable.getStackTrace());
+    } else {
+      log.error("Report failed, throwable is null {}", result.cause());
+    }
```

Comment:
Result can be null too

```suggestion
    if (throwable != null) {
      log.error("An exception was caught while building the report", throwable);
    } else if (result != null) {
      if (result.failed()) {
        log.error("Report construction failed: {}", result.cause());
      } else {
        log.info("Report built successfully");
      }
    }
```